src/modules/sub-items/components/table-view/table.view.component.js

```diff
@@ -21,6 +29,66 @@ export default class TableViewComponent extends Component {
         this.sortByPriority = this.sortByPriority.bind(this);
         this.renderItem = this.renderItem.bind(this);
         this.selectAll = this.selectAll.bind(this);
+        this.toggleColumnsToggler = this.toggleColumnsToggler.bind(this);
+        this.handleColumnsTogglerClose = this.handleColumnsTogglerClose.bind(this);
+
+        this._refColumnsTogglerButton = createRef();
+
+        this.headerLabels = {
+            name: Translator.trans(/*@Desc("Name")*/ 'items_table.header.name', {}, 'sub_items'),
+            modified: Translator.trans(/*@Desc("Modified")*/ 'items_table.header.modified', {}, 'sub_items'),
+            contentType: Translator.trans(/*@Desc("Content type")*/ 'items_table.header.content_type', {}, 'sub_items'),
+            priority: Translator.trans(/*@Desc("Priority")*/ 'items_table.header.priority', {}, 'sub_items'),
+            translations: Translator.trans(/*@Desc("Translations")*/ 'items_table.header.translations', {}, 'sub_items'),
+        };
+
+        this.state = {
+            isColumnsTogglerOpen: false,
+            columnsVisibility: this.getColumnsVisibilityFromLocalStorage(),
+        };
+    }
+
+    componentDidMount() {
+        document.addEventListener('click', this.handleColumnsTogglerClose, false);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('click', this.handleColumnsTogglerClose);
+    }
+
+    handleColumnsTogglerClose({ target }) {
+        if (!this.state.isColumnsTogglerOpen) {
+            return;
+        }
+
+        const isClickInsideColumnsToggler = target.closest('.c-table-view__columns-toggler-panel');
+        const isClickOnColumnsTogglerBtn = target.closest('.c-table-view__columns-toggler-btn');
+
+        if (!isClickInsideColumnsToggler && !isClickOnColumnsTogglerBtn) {
+            this.setState(() => ({
+                isColumnsTogglerOpen: false,
+            }));
+        }
+    }
+
+    getColumnsVisibilityFromLocalStorage() {
+        try {
```

Comment:
I'm not sure you need this `try ... catch` here, because you have control over everything here.